### PR TITLE
feat: ability to see constructor transaction traces and gas [APE-1097]

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

allow showing traces for deploy txns

Screenshot 2023-06-14 at 17 01 37

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Comment by antazoey** [comment]
- [ ] Show in gas report
- [ ] Ensure error-tracing works if failure happens in constructor

File: src/ape_ethereum/transactions.py [hunk 1]
```diff
@@ -154,11 +154,7 @@ def failed(self) -> bool:
 
     @cached_property
     def call_tree(self) -> Optional[CallTreeNode]:
-        if self.receiver:
-            return self.provider.get_call_tree(self.txn_hash)
-
-        # Not an function invoke
-        return None
```
Comment (sabotagebeats) [inline src/ape_ethereum/transactions.py:161]: Is this supposed to be indented?

File: tests/integration/cli/test_test.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
 
   Method +Times called +Min. +Max. +Mean +Median
  ─+
+  __init__ +\d +\d+ + \d+ + \d+ + \d+
```
Comment (sabotagebeats) [inline tests/integration/cli/test_test.py:16]: I'm not sure what these do

**Review by sabotagebeats — DISMISSED** [review]

File: tests/integration/cli/test_test.py (hunk 2 above)
Comment (antazoey) [reply tests/integration/cli/test_test.py:16]: it is regex that ensures the method has numeric values for all the gas columns with some spacing in between

File: src/ape_ethereum/transactions.py (hunk 1 above)
Comment (antazoey) [reply src/ape_ethereum/transactions.py:161]: we good

File: src/ape/pytest/fixtures.py [hunk 3]
```diff
@@ -157,7 +157,7 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
     def capture_range(self, start_block: int, stop_block: int):
         blocks = self.chain_manager.blocks.range(start_block, stop_block + 1)
-        transactions = [t for b in blocks for t in b.transactions if t.receiver and t.sender]
+        transactions = [t for b in blocks for t in b.transactions if t.sender]
```
Comment (antazoey) [inline src/ape/pytest/fixtures.py:160]: might be able to remove `if t.sender` as well, not sure why that is there... probably preventing impersonated TXNs from working actually... hmmmm..

- [ ] check this out

**Review by fubuloubu — APPROVED** [review]

**Pull request merged**